### PR TITLE
Merge 7.0.0 to dev

### PR DESCRIPTION
**Pull request by DellaBitta**
Merging code from main branch (6.16.0 release) to dev.
 - Resolved 123 merge conflicts.

### CODE REVIEW

**Review by vimanyu — APPROVED** [review]
nit: end of file new lines
LGTM otherwise.

File: firestore/src/ios/firebase_metadata_provider_desktop.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#ifndef FIREBASE_FIRESTORE_CLIENT_CPP_SRC_IOS_FIREBASE_METADATA_PROVIDER_DESKTOP_H_
+#define FIREBASE_FIRESTORE_CLIENT_CPP_SRC_IOS_FIREBASE_METADATA_PROVIDER_DESKTOP_H_
```
Comment (sunmou99) [inline firestore/src/ios/firebase_metadata_provider_desktop.h:2]: It seems the file name doesn't consist with the path. desktop files under ios path?

**Comment by DellaBitta** [comment]
> nit: end of file new lines
> LGTM otherwise.

Yep, the end of file problems came straight out of the internal repo.  I'm working on a CL to push buid fixes made here back into the internal repo.  I'll be sure to add new lines to these files.  Thanks!

**Comment by DellaBitta** [comment]
@googlebot I fixed it.

**Pull request merged**